pkg/liquidity-mining/test/ChildChainGaugeRegistry.test.ts

```diff
@@ -0,0 +1,135 @@
+import { ethers } from 'hardhat';
+import { Contract } from 'ethers';
+
+import { sharedBeforeEach } from '@balancer-labs/v2-common/sharedBeforeEach';
+import * as expectEvent from '@balancer-labs/v2-helpers/src/test/expectEvent';
+import { deploy, deployedAt } from '@balancer-labs/v2-helpers/src/contract';
+import { expect } from 'chai';
+import { ZERO_ADDRESS } from '@balancer-labs/v2-helpers/src/constants';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import Vault from '@balancer-labs/v2-helpers/src/models/vault/Vault';
+import { actionId } from '@balancer-labs/v2-helpers/src/models/misc/actions';
+
+describe('ChildChainGaugeRegistry', () => {
+  let gaugeImplementation: Contract;
+  let gaugeFactory: Contract;
+  let otherGaugeFactory: Contract;
+  let gauge: Contract;
+  let otherGauge: Contract;
+  let admin: SignerWithAddress, other: SignerWithAddress;
+  let registry: Contract;
+
+  const factoryVersion = JSON.stringify({
+    name: 'ChildChainGaugeFactory',
+    version: '1',
+    deployment: 'test-deployment',
+  });
+  const productVersion = JSON.stringify({
+    name: 'ChildChainGauge',
+    version: '0',
+    deployment: 'test-deployment',
+  });
+
+  before('setup signers', async () => {
+    [, admin, other] = await ethers.getSigners();
+  });
+
+  sharedBeforeEach('deploy gauge registry', async () => {
```

Comment:
Same as above, consider splitting the setup in smaller blocks if that makes sense to you.